### PR TITLE
Adding prop types oneOf value and updating the default returned marku…

### PR DESCRIPTION
**Pull request by rotemee**
…p for Text and Heading components.

## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/app/assets/js/ui/atoms/heading.js [hunk 1]
```diff
@@ -10,9 +10,19 @@ export default function Heading( props ) {
...
+	const classesString = Utils.arrayToClassName( classes );
+
+	if ( props.tag ) {
+		return React.createElement( tagName, {
+			className: classesString,
+		}, props.children );
+	}
+
+	return (
+		<h1 className={ classesString }>
```
Comment (matipojo) [inline core/app/assets/js/ui/atoms/heading.js:22]: Why not use a default `props.tag=h1`?

File: core/app/assets/js/ui/atoms/text.js [hunk 2]
```diff
@@ -3,19 +3,28 @@ export default function Text( props ) {
...
+	const classesString = Utils.arrayToClassName( classes );
+
+	if ( props.tag ) {
+		return React.createElement( props.tag, {
+			className: classesString,
+		}, props.children );
+	}
+
+	return (
+		<p className={ classesString }>
```
Comment (matipojo) [inline core/app/assets/js/ui/atoms/text.js:19]: `defaultProps.tag=p`

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/app/assets/js/ui/atoms/heading.js (hunk 1 above)
Comment (rotemee) [reply core/app/assets/js/ui/atoms/heading.js:22]: The storybook doesn't know how to get the component props when no element is returned. I understand it's unusual, but I think we need to consider it for the benefit of getting the component info automatically.

File: core/app/assets/js/ui/atoms/text.js (hunk 2 above)
Comment (rotemee) [reply core/app/assets/js/ui/atoms/text.js:19]: The storybook doesn't know how to get the component props when no element is returned. I understand it's unusual, but I think we need to consider it for the benefit of getting the component info automatically.

File: core/app/assets/js/ui/atoms/heading.js (hunk 1 above)
Comment (matipojo) [reply core/app/assets/js/ui/atoms/heading.js:22]: `const Element = () => React.createElement...

return <Element...`

Comment (rotemee) [reply core/app/assets/js/ui/atoms/heading.js]: Nice, fixed.

File: core/app/assets/js/ui/atoms/text.js (hunk 2 above)
Comment (rotemee) [reply core/app/assets/js/ui/atoms/text.js]: Fixed.

File: core/app/assets/js/ui/atoms/heading.js [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-		tagName = allowedTags.includes( props.tag ) ? props.tag : 'h1';
+		];
 
 	if ( props.variant ) {
 		classes.push( baseClassName + '-' + props.variant );
 	}
 
-	return React.createElement( tagName, {
-		className: classes.filter( ( classItem ) => classItem ).join( ' ' ),
+	const Element = () => React.createElement( 'h1', {
```
Comment (matipojo) [inline core/app/assets/js/ui/atoms/heading.js:11]:
```suggestion
	const Element = () => React.createElement( props.tag, {
```

File: core/app/assets/js/ui/atoms/heading.js [hunk 4]
```diff
@@ -22,8 +22,8 @@ Heading.propTypes = {
 		PropTypes.object,
 		PropTypes.arrayOf( PropTypes.object ),
 	] ).isRequired,
-	tag: PropTypes.string,
-	variant: PropTypes.string.isRequired,
+	tag: PropTypes.oneOf( [ 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' ] ),
+	variant: PropTypes.oneOf( [ 'display-1', 'display-2', 'display-3', 'display-4', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' ] ).isRequired,
 };
 
 Heading.defaultProps = {
```
Comment (matipojo) [inline core/app/assets/js/ui/atoms/heading.js:29]: `tag=h1`

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by kobizz — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**